### DISCUSSION TITLE
Skeletyl, ZMK, handwire with different pins compared to corne, cannot port as miryoku outboard

### DISCUSSION THREAD

**Discussion by gogamid** [manna-harbour/miryoku#349] [Q&A] [answered]
Hello,
I have finished the build of Skeletyl with directly wiring [columns and rows](https://github.com/gogamid/zmk-config-bastardkb/blob/bb42de21886b1e59169a862883a1a94ddb14dfe3/config/boards/shields/bastardkb/bastardkb.dtsi#L18C1-L21C72), and created [zmk-config-bastardkb](https://github.com/gogamid/zmk-config-bastardkb/tree/bastardkb/config/boards/shields/bastardkb) .

Then created bastardkb file in shields:
```
outboard_repository=gogamid/zmk-config-bastardkb
outboard_ref=bastardkb
outboard_from=config/boards/shields/bastardkb
outboard_to=boards/shields/bastardkb
```
then created bastardkb.keymap:
```
#include "../miryoku/custom_config.h"
#include "../miryoku/mapping/42/corne.h"
#include "../miryoku/miryoku.dtsi"
```

then when i run, build is successfull, but the layout is so wrong, not every key is working. So I assume somehow pinouts that i defined [here](https://github.com/gogamid/zmk-config-bastardkb/blob/bb42de21886b1e59169a862883a1a94ddb14dfe3/config/boards/shields/bastardkb/bastardkb_left.overlay#L4) are not considered. 

Can you help me to fix this?

my miryoku_zmk: https://github.com/gogamid/miryoku_zmk
my zmk-config: https://github.com/gogamid/zmk-config-bastardkb/tree/bastardkb

**Comment by manna-harbour** [comment, marked as answer]
When doing a handwire, the simplest approach is to wire according to the schematic of an existing keyboard that already has support in ZMK. For 3x5+3 I like to use the Corne schematic. That also can be used for 3x6+3. For the matrix details you can view the PCB files from the crkbd repo in KiCad, or refer to the ZMK shield definition.

If you prefer to make your own shield definition, first try building the default keymap directly without using Miryoku ZMK. If you have issues with that you can ask for help on the ZMK discord. If that is working but it's not correct in Miryoku ZMK it will be a helpful reference in diagnosing the issue.

When making your own shield definition, refer to the ZMK docs. You'll need to adjust the matrix pins, diode direction, and transform.
